pkgs/applications/blockchains/polkadot/default.nix

```diff
@@ -1,37 +1,52 @@
-{ stdenv
+{ callPackage
+, clang
 , fetchFromGitHub
-, rustPlatform
-, pkgconfig
-, openssl
+, llvmPackages
+, makeRustPlatform
+, protobuf
+, stdenv
 }:
-
-rustPlatform.buildRustPackage rec {
+let
+  mozillaOverlay = fetchFromGitHub {
+    owner = "mozilla";
+    repo = "nixpkgs-mozilla";
+    rev = "efda5b357451dbb0431f983cca679ae3cd9b9829";
+    sha256 = "11wqrg86g3qva67vnk81ynvqyfj0zxk83cbrf0p9hsvxiwxs8469";
+  };
+  mozilla = callPackage "${mozillaOverlay.out}/package-set.nix" { };
+  rustWasm = (mozilla.rustChannelOf { date = "2020-09-17"; channel = "nightly"; }).rust.override {
+    targets = [ "wasm32-unknown-unknown" ];
```

Comment:
I don't know how to go about doing this without using the mozilla overlay. We need to have a rustc that supports both the host target and also wasm32. The builtin nix rust support seems to only support defining one target environment.